File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Java/PhoneDisplay.java
public class PhoneDisplay implements Observer {
    private Subject weatherStation;

    public PhoneDisplay(Subject weatherStation) {
        this.weatherStation = weatherStation;
        weatherStation.addObserver(this);
    }

    @Override
    public void update(float temperature) {
        System.out.println("Phone Display: Temperature updated to " + temperature + "°C");
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Cpp/src/PhoneDisplay.cpp
#include "../include/PhoneDisplay.h"

PhoneDisplay::PhoneDisplay(Subject* subject) : subject(subject) {
    subject->addObserver(this);
}

void PhoneDisplay::update(float temperature) {
    std::cout << "Phone Display: Temperature updated to " << temperature << "°C\n";
}
